### PR TITLE
Added notification const NOTIFICATION_WM_ABOUT

### PR DESCRIPTION
**Pull request by marcelofg55**
I've added that NOTIFICATION_WM_ABOUT to fix the "About godot" button on OS X, since it's a notification it could be used on games as well.

Before this PR:
screen shot 2017-08-11 at 11 50 36 am

After this PR:
screen shot 2017-08-11 at 12 01 46 pm

### CODE REVIEW

File: scene/main/scene_tree.cpp [hunk 1]
```diff
@@ -701,6 +702,15 @@ void SceneTree::_notification(int p_notification) {
 
 		} break;
 
+		case NOTIFICATION_WM_ABOUT: {
+
+#ifdef TOOLS_ENABLED
+			EditorNode::get_singleton()->show_about();
```
Comment (akien-mga) [inline scene/main/scene_tree.cpp]: That won't work if receiving the notification while playing the game with a `tools=yes` binary. You should likely check the editor hint in the `#ifdef`, and if not also default to propagating the notification.

Comment (akien-mga) [inline scene/main/scene_tree.cpp]: Something like this:
```
#ifdef TOOLS_ENABLED
	if (get_tree()->is_editor_hint()) {
		EditorNode::get_singleton()->show_about();
	} else {
#endif
		get_root()->propagate_notification(p_notification);
#ifdef TOOLS_ENABLED
	}
#endif
```

Comment (akien-mga) [inline scene/main/scene_tree.cpp]: Or maybe this is more readable:
```
#ifdef TOOLS_ENABLED
	if (get_tree()->is_editor_hint()) {
		EditorNode::get_singleton()->show_about();
	} else
#endif
	{
		get_root()->propagate_notification(p_notification);
	}
```

Comment (akien-mga) [inline scene/main/scene_tree.cpp]: (or maybe not, more bug prone :))

Comment (marcelofg55) [inline scene/main/scene_tree.cpp]: Nice catch, I missed that. Just changed it and fixed it :)

**Comment by marcelofg55** [comment]
@akien-mga I've just realized that NOTIFICATION_TRANSLATION_CHANGED (id=10) and this NOTIFICATION_WM_ABOUT (id=11) conflict with NOTIFICATION_ENTER_TREE (id=10) and NOTIFICATION_EXIT_TREE (id=11). Should I change NOTIFICATION_TRANSLATION_CHANGED and NOTIFICATION_WM_ABOUT to 90/91?

**Comment by akien-mga** [comment]
Yes, sounds good. Please add a comment to explain the discrepancy.

**Comment by toger5** [comment]
what does WM stand for? is that the best name?

**Comment by marcelofg55** [comment]
@toger5 WM stands for Window Manager here.

**Comment by marcelofg55** [comment]
@akien-mga I've just made the ids change, and left a note with it. As an idea, maybe each class that uses NOTIFICATION_* ids could each allocate a 100 range? (like MainLoop ids goes from 0-100, Node would go from 101-200 and so on)

File: core/os/main_loop.cpp [hunk 2]
```diff
@@ -54,6 +54,7 @@ void MainLoop::_bind_methods() {
 	BIND_CONSTANT(NOTIFICATION_WM_QUIT_REQUEST);
 	BIND_CONSTANT(NOTIFICATION_WM_UNFOCUS_REQUEST);
 	BIND_CONSTANT(NOTIFICATION_OS_MEMORY_WARNING);
+	BIND_CONSTANT(NOTIFICATION_WM_ABOUT);
```
Comment (akien-mga) [inline core/os/main_loop.cpp]: Should `NOTIFICATION_TRANSLATION_CHANGED` be bound too?

**Comment by akien-mga** [comment]
FYI, we discussed it on IRC with @reduz and @bojidar-bg and it seems those notifications are actually not conflicting:
```
<Akien> reduz: FYI, NOTIFICATION_TRANSLATION_CHANGED (id 10) used to conflict with NOTIFICATION_ENTER_TREE (id 10). This PR will fix it, but we should maybe think about ensuring notification enums don't overlap. #10254
<bojidar_bg> Akien: BTW, SceneTree/Mainloop does not inherit Node, so overlap shouldn't matter much
<reduz> Akien: ah thats not a conflict, as bojidar_bg says, they are not in the same inheritance chain. MainLoop redirects those using NODE notifications
<bojidar_bg> still, NOTIFICATION_ enums should not conflict if possible :)
<reduz> i guess
```

**Comment by marcelofg55** [comment]
Even if they don't overlap, won't the numbers conflict when _notification(what) gets called on a GDscript?

**Comment by bojidar-bg** [comment]
@marcelofg55 You can't have a GDScript that is both a SceneTree and a Node, unless I'm missing something

**Comment by bojidar-bg** [comment]
Aww, my bad, actually, they shouldn't conflict, since SceneTree is propagating the notification throughout the whole tree.

**Comment by akien-mga** [comment]
> Aww, my bad, actually, they shouldn't conflict, since SceneTree is propagating the notification throughout the whole tree.

You mean they *will* conflict and thus should be distinct, right?

File: core/os/main_loop.cpp (hunk 2 above)
Comment (marcelofg55) [inline core/os/main_loop.cpp]: That's true, I'll add it

**Comment by bojidar-bg** [comment]
Yes

**Comment by bruvzg** [comment]
@marcelofg55 works in editor and projects, but segfaults in Project Manager.

```
Process:               godot.osx.tools.64 [76893]
Path:                  /Volumes/VOLUME/*/godot.osx.tools.64
Identifier:            godot.osx.tools.64
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        bash [74026]
Responsible:           godot.osx.tools.64 [76893]
User ID:               501

Date/Time:             2017-08-22 17:42:47.564 +0300
OS Version:            Mac OS X 10.12.6 (16G29)
Report Version:        12
Anonymous UUID:        7D79862A-6DAA-D82C-F878-BCAF4B670F2C


Time Awake Since Boot: 39000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00000000000008d4
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0x8d4:
--> 
    __TEXT                 00000001054da000-00000001080e3000 [ 44.0M] r-x/rwx SM=COW  /Volumes/VOLUME/*/*.64

Application Specific Information:
Performing @selector(showAbout:) from sender NSMenuItem 0x7f8990c1be60

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   godot.osx.tools.64            	0x0000000105d2dd38 EditorNode::_menu_option_confirm(int, bool) + 88 (editor_node.cpp:1858)
1   godot.osx.tools.64            	0x000000010639528c EditorNode::show_about() + 28 (editor_node.h:687)
2   godot.osx.tools.64            	0x0000000106395229 SceneTree::_notification(int) + 489 (scene_tree.cpp:635)
3   godot.osx.tools.64            	0x00000001063aa9a7 SceneTree::_notificationv(int, bool) + 151 (scene_tree.h:75)
4   godot.osx.tools.64            	0x0000000106fea892 Object::notification(int, bool) + 50 (object.cpp:956)
5   godot.osx.tools.64            	0x00000001054db653 -[GodotApplicationDelegate showAbout:] + 83 (os_osx.mm:143)
6   libsystem_trace.dylib         	0x00007fff9044b3a7 _os_activity_initiate_impl + 53
7   com.apple.AppKit              	0x00007fff788cc721 -[NSApplication(NSResponder) sendAction:to:from:] + 456
8   com.apple.AppKit              	0x00007fff7839f666 -[NSMenuItem _corePerformAction] + 324
9   com.apple.AppKit              	0x00007fff7839f3d2 -[NSCarbonMenuImpl performActionWithHighlightingForItemAtIndex:] + 114
10  libsystem_trace.dylib         	0x00007fff9044b3a7 _os_activity_initiate_impl + 53
11  com.apple.AppKit              	0x00007fff78428955 -[NSMenu performActionForItemAtIndex:] + 121
12  com.apple.AppKit              	0x00007fff784288cc -[NSMenu _internalPerformActionForItemAtIndex:] + 94
13  com.apple.AppKit              	0x00007fff784286f9 -[NSCarbonMenuImpl _carbonCommandProcessEvent:handlerCallRef:] + 107
14  com.apple.AppKit              	0x00007fff782df580 NSSLMMenuEventHandler + 986
15  com.apple.HIToolbox           	0x00007fff79b8dd85 DispatchEventToHandlers(EventTargetRec*, OpaqueEventRef*, HandlerCallRec*) + 1708
16  com.apple.HIToolbox           	0x00007fff79b8cff6 SendEventToEventTargetInternal(OpaqueEventRef*, OpaqueEventTargetRef*, HandlerCallRec*) + 428
17  com.apple.HIToolbox           	0x00007fff79ba2d14 SendEventToEventTarget + 40
18  com.apple.HIToolbox           	0x00007fff79bef3e6 SendHICommandEvent(unsigned int, HICommand const*, unsigned int, unsigned int, unsigned char, void const*, OpaqueEventTargetRef*, OpaqueEventTargetRef*, OpaqueEventRef**) + 411
19  com.apple.HIToolbox           	0x00007fff79c1a9fb SendMenuCommandWithContextAndModifiers + 59
20  com.apple.HIToolbox           	0x00007fff79c1a9aa SendMenuItemSelectedEvent + 188
21  com.apple.HIToolbox           	0x00007fff79c1a87d FinishMenuSelection(SelectionData*, MenuResult*, MenuResult*) + 96
22  com.apple.HIToolbox           	0x00007fff79c1b2b7 MenuSelectCore(MenuData*, Point, double, unsigned int, OpaqueMenuRef**, unsigned short*) + 702
23  com.apple.HIToolbox           	0x00007fff79c1af66 _HandleMenuSelection2 + 460
24  com.apple.AppKit              	0x00007fff782cf368 _NSHandleCarbonMenuEvent + 239
25  com.apple.AppKit              	0x00007fff78549702 _DPSEventHandledByCarbon + 54
26  com.apple.AppKit              	0x00007fff788ca0c5 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 963
27  godot.osx.tools.64            	0x00000001054e45d5 OS_OSX::process_events() + 101 (os_osx.mm:1781)
28  godot.osx.tools.64            	0x00000001054e4725 OS_OSX::run() + 149 (os_osx.mm:1825)
29  godot.osx.tools.64            	0x00000001054e8d12 main + 722 (godot_main_osx.mm:84)
30  libdyld.dylib                 	0x00007fff90219235 start + 1
```

**Comment by akien-mga** [comment]
Indeed, the project manager is considered as `is_editor_hint() == true`, but EditorNode does not exist at this stage.

**Comment by marcelofg55** [comment]
Just submitted #10556, should fix the crash.

**Pull request merged**